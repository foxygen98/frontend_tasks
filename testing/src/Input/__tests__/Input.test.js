import React from 'react';
import {render, screen, fireEvent, cleanup} from '@testing-library/react';
import Input from '../Input';

describe('<Input />', () => {
  afterEach(cleanup);

  it('should render input', () => {
    const wrapper = render(<Input value='value' />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should change', () => {
    render(<Input value="value" />);
    fireEvent.change(screen.getByRole('textbox'), { 
      target: { value: 'changed' }, 
    });
    expect(screen.getByDisplayValue('changed')).toBeInTheDocument();
  });
})
