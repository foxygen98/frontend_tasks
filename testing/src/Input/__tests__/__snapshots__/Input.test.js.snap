// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render input 1`] = `
<DocumentFragment>
  <input
    value="value"
  />
</DocumentFragment>
`;
